### PR TITLE
fix(scripts/RuinsOfAhnQiraj): Improve Buru

### LINKED ISSUE #12418: [AQ20] Buru the Gorger - Buru Eggs despawn & respawn
### Current Behaviour

- [ ] Eggs respawn and despawn timer needs to be readjusted. Vmangos' respawn is scripted in cpp with a 120s respawn timer, I believe. Currently it's set to 2h. In retail the respawn timer is also 2min

### Expected Blizzlike Behaviour

- Eggs should be able to respawn during the fight.
- Eggs should not respawn after the fight.
- Eggs should respawn if the boss resets.
- Eggs despawn after around 5s, seen in retail and WotLK footage.
- During Phase 2 transition all eggs burst, and do not respawn afterwards.

### Source

_No response_

### Steps to reproduce the problem

1. `.go c id 15370`
2. engage boss, kill eggs.
3. `.damage 600000` on boss, see all eggs burst.
4. kill boss, see that eggs do not respawn.

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/cc52712ac12e3d6a16aabcbd13807924c64aeb1e

### Operating system

Windows 10

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by Si1ker**
Originated from #12061

## Changes Proposed:
- Cleanup/refactor
- Schedule EVENT_CREEPING_PLAGUE
- Adjust timer for EVENT_CREEPING_PLAGUE
- Immune to Taunt
- Let "Creeping Plague" tick 3 times before stacking up
- Remove EVENT_DISMEMBER on Phase2  |  https://youtu.be/D0nktm78pII?t=375
- Make all visual spell effects triggered to skip unit egg death
- Remove all 'Creeping Plague' stacks on Buru death
- Make the remaining eggs hatch at Phase 2 (won't respawn)  |  https://youtu.be/TyRheH9Cys0?t=382
- Disable experience gain on "Buru Egg" and "Hive'Zara Hatchling"
- Disable visual effects on eggs at "Phase2 forced hatch"
- Despawn "Hive'Zara Hatchling" after 10sec if out of combat
- Closes #12418

## Tests Performed:

- Tested

## How to Test the Changes:

1. Go to AQ20
2. Engage Buru

## Known Issues and TODO List:

* Buru missing damage reduction in Phase 1  |  [idea](https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Northrend/Ulduar/Ulduar/boss_hodir.cpp#L335)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Si1ker** [comment]
ready

**Review by acidmanifesto — DISMISSED** [review]

**Comment by Gultask** [comment]
Tested.

>Root egg properly - avoid reset

What was the issue here?

Other than my question everything looks good.

File: data/sql/updates/pending_db_world/rev_1655864042959599100.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+UPDATE `creature_template` SET `flags_extra`=`flags_extra`|64 WHERE `entry` IN (15514,15521);
+
+-- New smartAI - despawn
+UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`=15521;
+DELETE FROM `smart_scripts` WHERE `entryorguid`= 15521 AND `source_type`= 0 AND `id`= 0;
+INSERT INTO `smart_scripts` (`entryorguid`, `event_type`, `event_flags`, `event_param1`, `event_param2`, `action_type`, `action_param1`, `target_type`, `comment`) VALUES 
+(15521, 1, 1, 10000, 10000, 41, 500, 1, 'Hive\'Zara Hatchling - Out of Combat - Despawn (No Repeat)');
+
+-- Root egg properly - avoid reset
+DELETE FROM `creature_template_movement` FROM `CreatureId`=15514;
```
[the remark this answers is not in the document]
Comment (Si1ker) [reply data/sql/updates/pending_db_world/rev_1655864042959599100.sql]: fixed

**Comment by Si1ker** [comment]
> 

reset problems (aka Ragnaros incident).

**Comment by Si1ker** [comment]
> Tested.
> 
> > Root egg properly - avoid reset
> 
> What was the issue here?
> 
> Other than my question everything looks good.

reset problems (aka Ragnaros incident).

**Review by Nefertumm — APPROVED** [review]

**Comment by Si1ker** [comment]
## Known Issues and TODO List:


* Buru missing damage reduction in Phase 1  |  [idea](https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Northrend/Ulduar/Ulduar/boss_hodir.cpp#L335)

**Pull request merged**